packages/phpstan-rules/src/Rules/PreventDuplicateClassMethodRule.php

```diff
@@ -180,11 +180,11 @@ private function validateDuplication(
                 continue;
             }
 
-            if (in_array($duplicationPlaceholder, $this->reportedClassWithMethodDuplicate, true)) {
+            if (in_array($duplicationPlaceholder, self::$reportedClassWithMethodDuplicate, true)) {
                 continue;
             }
 
-            $this->reportedClassWithMethodDuplicate[] = $duplicationPlaceholder;
+            self::$reportedClassWithMethodDuplicate[] = $duplicationPlaceholder;
```

Comment:
`static` is probably not good idea as it breaks service design. This work around should be replace with clean solution.

Why is even reported twice? Other rules are reported just once.

It seems like this rule is registered twice.

Reply:
I don't have idea, in this repository, it only reported once if I clean up phpstan.neon ignored message, while in rector, it reported twice, the rule only registered in `packages/phpstan-rules/config/static-rules.neon`.

Is it probably related with some autoload changed in rector?

Reply:
@TomasVotruba  In rector, I found it registered twice, in : 

- utils/phpstan-extensions/config/rector-rules.neon
- phpstan.neon load  `vendor/symplify/phpstan-rules/config/static-rules.neon`

is that probably cause this?